tq/transfer_queue.go

```diff
@@ -307,7 +307,7 @@ func (q *TransferQueue) enqueueAndCollectRetriesFor(batch batch) (batch, error)
 	}
 
 	q.useAdapter(bRes.TransferAdapterName)
-	q.startProgress.Do(q.meter.Start)
```

Comment:
Before we resume the progress meter, we have two opportunities to bail out early: on L302 and L306. I think both of those are fine, since we'd exit this function early, and the progress meter would get resumed the next time a new batch was enqueued. There'd be no issues with calling `Pause()` twice, since there's a `CompareAndSwap<T>` operation guarding the channel send into `p.finished`.

@technoweenie would you mind double checking my thinking on this?